docs/enhancements/metallb-operator.md

```diff
@@ -0,0 +1,205 @@
+
+- Which is the *internal network*?
+- What is the IP on the **external network** to use for the API VIP?
+- What is the IP on the **external network** to use for the Ingress?
+- What routes (if any) should be used for the **external** API and Ingress endpoints?
+
+### Identifying Internal Network
+
+Assisted installer already has mechanims to know which is the internal network. Internally this
+network is called the `MachineNetwork`. In case of SNO, the `MachineNetwork` is required when defining
+the ACI. In the case of a multinode deployment, it is inferred from the API VIP whereas in the case
+of dual stack this information is required the same way as in the SNO case.
+
+Regardless, no changes should be required in assisted installer to identify the internal network.
+
+### Identifying External Network
+
+This enhancement proposes requesting the user for the following information in order to prepare the
+cluster for relocation:
+
+- External API VIP: This is an IP on the **external** network
+- External Ingress IP: This is an IP on the **external** network
+- External base domain: A domain to use for the **external** traffic (optional).
+
+Unlike for the **internal** network, this enhancement proposes to always infer the subnet of the
+external network from the External API VIP and the External Ingress IP provided by the user. There
+are utilities in Assisted Installer to do this already.
+
+Some requirements:
+
+- The **external** API VIP and Ingress IP must be on the same subnet.
+- The host must be connected to this network and the network must be routable
+- These 2 IPs must be optional. The IPAddressPool and Service won't be configured if this information is not provided
+- Either both are IPs are provided or none.
+- The *internal* network must be configured as a static network to prevent the MachineNetwork to
+change and, therefore, the IP of each of the nodes.
+
+### User Stories
+
+#### Story 1
+
+As a hardware provider, I want to be able to deliver an OCP appliance that can be shipped to a
+different location and reconnected.
+
+#### Story 2
+
+As a datacenter administrator, I want to be able to install an OCP cluster whose IP can be
+reconfigured as well as its domain.
+
+### Implementation Details/Notes/Constraints [optional]
+
+1. It's currently not possible to enable operators when using KubeAPI. Implementing this feature,
+without enabling it through KubeAPI will limit the scope of it for environments that use ZTP. It
+should be considered to allow enabling operators through KubeAPI either as part of this enhancement
+or a separate one.
+2. Assisted Installer doesn't have a way to configure an operator. Operators can be enabled or
+disabled. The rest of the information they take from the infrastructure data that has been
+collected. For this enhancement, we need to either introduce the concept of External API VIP and
+External Ingress IP to the cluster or allow for specifying operator-specific configurations.
```

Comment:
The API has a way to define and pass operator-specific parameters, but none of the currently-supported operators use this.  I'm wondering if we should expose this feature as "install metalLB" or "make this cluster relocatable".  I'm leaning toward the second option.

Reply:
> I'm wondering if we should expose this feature as "install metalLB" or "make this cluster relocatable". I'm leaning toward the second option.

Agreed! Big part of the reason is that the configurations and validations we would expose work under the assumption that we are building a relocatable cluster. Otherwise, we would not need any deployment-time configurations but rather just install MetalLB